pykafka/simpleconsumer.py

```diff
@@ -828,7 +829,18 @@ def enqueue_messages(self, messages):
         :type messages: Iterable of :class:`pykafka.common.Message`
         """
         for message in messages:
-            if message.offset != self.next_offset:
+            # HAck to deal with compacted topic not returning earliest offset
+            # correctly
+            if not self._consumed_first_msg and message.offset != self.next_offset:
+                log.info("Detected compacted topic. "
+                          "Setting earliest offest to (%s) "
+                          "not equal to next_offset (%s)",
+                          message.offset, self.next_offset)
+                self.next_offset = message.offset
+                self._consumed_first_msg = True
```

Comment:
Can you help me understand why this variable is necessary? Since compacted topics can have sparse offsets, won't it always be necessary to set `next_offset` here?